src/zerocoin_v3.cpp

```diff
@@ -765,8 +766,9 @@ void CSigmaState::Containers::RemoveSpend(Scalar const & serial) {
     spend_info_container::const_iterator iter = usedCoinSerials.find(serial);
     if (iter != usedCoinSerials.end()) {
         spendMetaInfo[iter->second.coinGroupId][iter->second.denomination] -= 1;
+        CSpendCoinInfo tmpSpendInfo(iter->second);
         usedCoinSerials.erase(iter);
-        CheckSurgeCondition(iter->second.coinGroupId, iter->second.denomination);
+        CheckSurgeCondition(tmpSpendInfo.coinGroupId, tmpSpendInfo.denomination);
```

Comment:
Removal of spend can never lead to surge condition

Reply:
But it can remove the condition.

Reply:
Point taken but just to prevent forks when surging